### PR TITLE
Decrease disco volume by half

### PR DESCRIPTION
**Pull request by Mooshimi**
## About The Pull Request
-sets max disco ball volume to 50/100 (which is still a bit loud, fair for max volume)
-as disco is a subset of jukebox, jukebox is also set to this volume range

## Why It's Good For The Game
-our ears are bleeding
-please why

## Changelog
:cl:
balance: lowered disco ball's max volume
/:cl:

### CODE REVIEW

**Comment by Mooshimi** [comment]
A note: The UI volume knob I have no idea how to do. This just caps it at 50% volume visually. If someone could tweak the knob also to where it's max is 50 that would be smackin.

**Comment by SkeletalElite** [comment]
> A note: The UI volume knob I have no idea how to do. This just caps it at 50% volume visually. If someone could tweak the knob also to where it's max is 50 that would be smackin.

make emagged able to go over

**Comment by Mothblocks** [comment]
just change the tgui, its probably in a file named like Jukebox.ts

File: code/game/machinery/dance_machine.dm [hunk 1]
```diff
@@ -166,10 +166,11 @@
 				volume = 0
 				return TRUE
 			else if(new_volume == "max")
-				volume = 100
+				volume = 50
 				return TRUE
 			else if(text2num(new_volume) != null)
-				volume = text2num(new_volume)
+				///force volume back to 50 or the lower value
+				volume = (text2num(new_volume))/2
```
Comment (LemonInTheDark) [inline code/game/machinery/dance_machine.dm:173]: I don't really care about this change, but don't do this.
you should change the actual optional range for new_volume, not just manually scale it down later. that's really fragile.

File: code/game/machinery/dance_machine.dm [hunk 2]
```diff
@@ -166,10 +166,11 @@
 				volume = 0
 				return TRUE
 			else if(new_volume == "max")
-				volume = 100
+				volume = 50
```
Comment (Mothblocks) [inline code/game/machinery/dance_machine.dm:169]:
```suggestion
				volume = initial(volume)
```

File: code/game/machinery/dance_machine.dm (hunk 1 above)
Comment (Mothblocks) [inline code/game/machinery/dance_machine.dm:173]: don't do this, fix the tgui instead

**Review by Mothblocks — CHANGES_REQUESTED** [review]

File: code/game/machinery/dance_machine.dm [hunk 3]
```diff
@@ -166,10 +166,10 @@
 				volume = 0
 				return TRUE
 			else if(new_volume == "max")
-				volume = 100
+				volume = initial(volume)
 				return TRUE
 			else if(text2num(new_volume) != null)
-				volume = text2num(new_volume)
+				volume = (text2num(new_volume))
```
Comment (Mothblocks) [inline code/game/machinery/dance_machine.dm:172]: remove the parens now that you dont need it anymore

Comment (Mooshimi) [reply code/game/machinery/dance_machine.dm:172]: good eye

**Review by Mothblocks — APPROVED** [review]

**Comment by CourageTheCat** [comment]
Now Make Jukeboxes Craftable

**Comment by LemonInTheDark** [comment]
no

**Comment by Mooshimi** [comment]
> no

toward the crafting or the volume change

**Comment by LemonInTheDark** [comment]
Volume change, I wouldn't do mothblocks dirty like that. If I want 200db of caramelldansen in game I can just vv the thing's volume or bully an admin into doing it

**Comment by Mooshimi** [comment]
> Now Make Jukeboxes Craftable

I had a pr made up to add crafting for it, but they tell me jukeboxes cause lag with the way they send noises to your client, so they don't want them craftable. Sorry!

**Comment by LemonInTheDark** [comment]
It's all good. I only know that's a problem because I've seen people like me rant about it before

**Review by BallastMonster — COMMENTED** [review]
ACCIDENTAL REVIEW

**Review by Fikou — CHANGES_REQUESTED** [review]
i like disco

**Pull request merged**